custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js

```diff
@@ -195,6 +195,43 @@ describe('Download Directive', function () {
 
             assert.equal(expected, result);
         });
+
+        it('tests isDistrictOrBelowSelected - state selected', function () {
+            controller.selectedLocations = ['state', 'all'];
+            var result = controller.isDistrictOrBelowSelected();
+            assert.isFalse(result);
+        });
+
+        it('tests isDistrictOrBelowSelected - district selected', function () {
+            controller.selectedLocations = ['state', 'district'];
+            var result = controller.isDistrictOrBelowSelected();
+            assert.isTrue(result);
+        });
+
+        it('tests isBlockOrBelowSelected - district selected', function () {
+            controller.selectedLocations = ['state', 'district', 'all'];
+            var result = controller.isBlockOrBelowSelected();
+            assert.isFalse(result);
+        });
+
+        it('tests isBlockOrBelowSelected - block selected', function () {
+            controller.selectedLocations = ['state', 'district', 'block'];
+            var result = controller.isBlockOrBelowSelected();
+            assert.isTrue(result);
+        });
+
+        it('tests isAWCsSelected - AWC not selected', function () {
+            controller.selectedAWCs = [];
+            var result = controller.isAWCsSelected();
+            assert.isFalse(result);
+        });
+
+        it('tests isAWCsSelected - AWC selected', function () {
+            controller.selectedAWCs = ['awc_1', 'awc_@'];
```

Comment:
Should this be `awc_2`?

Reply:
@emord The strings in the list are not important because in ```isAWCsSelected``` function we check only list length. But I can change this if you want.